### PR TITLE
[sk] Added data exporters + templates

### PR DESCRIPTION
**Pull request by skunichetty**
# Summary
Added data exporters + templates. These blocks consume a data frame and export to the specified data source.

This involved refactoring the `data_loaders` library to support more general `io` functions (hence being renamed to the `io` library). Now the `data_loader` classes include a function `export` that takes in a Pandas data frame alongside other other parameters and exports that data frame to the specified source.

One new dependency was added: SQLAlchemy (the ORM part of the library is unused, but the SQL Engine that comes with this library is used in order to manage writing to PostgreSQL databases with the correct data types and for easy integration with Pandas).

# Tests
- Unit tests were added to confirm that the correct templates were loaded from file to create a data exporter block
- Testing was done to confirm that exporting data to the data sources is performed successfully.

cc:
@wangxiaoyou1993

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Are you still using psycopg2-binary package? If not, you can remove it from requirements.txt

**Comment by skunichetty** [comment]
I am still using psycopg2, just not implicitly. SQLAlchemy only provides an engine to generate SQL requests, psycopg2 is still required to interface with the database.

File: mage_ai/io/bigquery.py [hunk 1]
```diff
@@ -59,6 +59,44 @@ def load(self, query_string: str, **kwargs) -> DataFrame:
...
+        if_exists: str = 'replace',
+        **configuration_params,
+    ) -> None:
+        """
+        Exports a data frame to a Google BigQuery warehouse. If the table to export to exists,
+        the table will be appended to the end of the table.
+
+        Args:
+            df (DataFrame): Data frame to export
+            table_name (str): Name of the table to export the data frame to. If this table exists,
```
[the remark this answers is not in the document]
Comment (skunichetty) [reply mage_ai/io/bigquery.py]: Changed the docs to better reflect Google's terminology.

File: mage_ai/io/snowflake.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            if_exists (str): Specifies export policy if table exists. Either
+                - `'fail'`: throw an error.
+                - `'replace'`: drops existing table and creates new table of same name.
+                - `'append'`: appends data frame to existing table. In this case the schema must match the original table.
+            Defaults to `'append'`.
+            **kwargs: Additional arguments (such as `database` or `schema` to specify the database and schema to use)
+        """
+        exists = False
+        with self._ctx.cursor() as cur:
+            cur.execute(f'SHOW TABLES LIKE \'{table_name}\'')
```
[the remark this answers is not in the document]
Comment (skunichetty) [reply mage_ai/io/snowflake.py]: Fixed by adding database and schema parameters to export function.

File: mage_ai/io/snowflake.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        Exports a Pandas data frame to a Snowflake warehouse based on the table name. If table doesn't
+        exist, the table is automatically created.
+
+        Args:
+            df (DataFrame): Data frame to export to a Snowflake warehouse.
+            table_name (str): Name of the table to export the data to (excludes database name, schema name).
+            if_exists (str): Specifies export policy if table exists. Either
+                - `'fail'`: throw an error.
+                - `'replace'`: drops existing table and creates new table of same name.
+                - `'append'`: appends data frame to existing table. In this case the schema must match the original table.
```
[the remark this answers is not in the document]
Comment (skunichetty) [reply mage_ai/io/snowflake.py]: Fixed.

File: mage_ai/io/snowflake.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
+from mage_ai.io.base import BaseSQL
+from pandas import DataFrame
+from snowflake.connector import connect
+from snowflake.connector.pandas_tools import write_pandas
+
+
+class Snowflake(BaseSQL):
+    """
+    Loads data from a Snowflake data warehouse.
```
[the remark this answers is not in the document]
Comment (skunichetty) [reply mage_ai/io/snowflake.py]: Fixed (both here and in all other data IO handlers)

File: mage_ai/io/bigquery.py [hunk 5]
```diff
@@ -59,6 +59,44 @@ def load(self, query_string: str, **kwargs) -> DataFrame:
...
+    def export(
+        self,
+        df: DataFrame,
+        table_name: str,
+        if_exists: str = 'replace',
+        **configuration_params,
+    ) -> None:
+        """
+        Exports a data frame to a Google BigQuery warehouse. If the table to export to exists,
+        the table will be appended to the end of the table.
```
[the remark this answers is not in the document]
Comment (skunichetty) [reply mage_ai/io/bigquery.py]: Yes, fixed now (I forgot to update the docs after making the change on my end).

File: mage_ai/io/bigquery.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (skunichetty) [reply mage_ai/io/bigquery.py]: Okay, added this line both to docs and to template to make ID definition clearer.

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**